Enhance filter column change logic and use R square values over Pearson Correlation

Arun-bioinformatics opened the pull request:
- Removed redundant conditions related to 'rescued' variants.
- Added R² correlation calculations for both all samples and PASS-related samples in `create_interactive_correlation_plot`.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/5)
<!-- Reviewable:end -->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Interactive correlation plots now display R² for all samples and for a PASS-related subset, with titles showing both values.

- Refactor
  - Filtering treats PASS and rescued variants as PASS-related; colour categories simplified by removing rescued-specific labels and exposing new category labels.
  - Switched from Pearson to R² metrics for overall and subset correlations.

- Bug Fixes
  - Added error handling when insufficient PASS-related variants exist for R² calculation.

- Documentation
  - Updated notes and examples to reflect PASS-inclusive logic and R² outputs.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
Enhance filter column change logic and use R square values over Pearson Correlation

- Removed redundant conditions related to 'rescued' variants.
- Added R² correlation calculations for both all samples and PASS-related samples in `create_interactive_correlation_plot`.

RSWilson1 reviewed (changes requested):
> @RSWilson1 reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/5)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @Arun-bioinformatics)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 17 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/5#-Ob2EtQ5-P-tu0d5NP9S:-Ob2EtQ5-P-tu0d5NP9T:b-8xnwj3) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/90cf430665bade23458e04ce705efb4d7ed2183e/DI-2086/correlation_plots_pdf_html.py#L17)):*
> ```Python
>         Example pair values: 
>             - 'PASS' -> 'PASS'
>             - 'rescued' -> 'PASS'
> ```

rescued always is associated with another filter i.e. LowSupport, Blacklist etc.. never on it's own so probably best to update these example scenarios which are just rescued.


<!-- Sent from Reviewable.io -->

RSWilson1 reviewed (commented):
> @RSWilson1 reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/5)* status: all files reviewed, 1 unresolved discussion (waiting on @Arun-bioinformatics)



<!-- Sent from Reviewable.io -->

Arun-bioinformatics commented on DI-2086/correlation_plots_pdf_html.py:
```diff
@@ -186,19 +182,28 @@ def create_interactive_correlation_plot(csv_file, output_path):
         print("The unique values for FILTER_truth", df["FILTER_truth"].unique())
         print("The unique values for FILTER_query", df["FILTER_query"].unique())
 
-        # Calculate correlation
-        correlation = df["VAF_truth"].corr(df["VAF_query"])
+        # Calculate correlation of all samples
+        r2_total = r2_score(df["VAF_truth"], df["VAF_query"])
+        print(f"R² correlation: {r2_total:.4f}")
+        
+        # Calculate the correlation of only PASS related samples
+        subset_df = df[df["FILTER_change"].isin(
+            ["No change PASS",
+             "Change from PASS to EXCLUDE",
+             "Change from EXCLUDE to PASS",
+             "Variant added PASS",
+             "Variant removed PASS"])]
+        r2_subset = r2_score(subset_df["VAF_truth"], subset_df["VAF_query"])
+        print(f"R² correlation (PASS related): {r2_subset:.4f}")
```
@coderabbitai[bot], if the python script was part of a bash for loop, would your proposed fix break the for loop?
<!-- Reviewable comment -Ob2IT_v-H-SOJH-3fPs:b-aswvv4 -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics replied:
I have included this step now
<!-- Reviewable comment -Ob2LOtMCebUMOkZE3Tx:bb0my1o -->
<!-- Sent from Reviewable.io -->

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/5)* status: all files reviewed, 1 unresolved discussion (waiting on @Arun-bioinformatics)



<!-- Sent from Reviewable.io -->

Commit:
Address Rob and coderabbit's comments:
- Changed 'No change PASS' cases to not have 'rescued' only examples
-  Added error handling step if the subset of PASS related variants happens to be less than 2 variants.
- Removed trailing white spaces

Arun-bioinformatics reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/5)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @RSWilson1)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 17 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/5#-Ob2EtQ5-P-tu0d5NP9S:-Ob2LMmgClAF9qxSKzV_:b-bu91xm) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/90cf430665bade23458e04ce705efb4d7ed2183e/DI-2086/correlation_plots_pdf_html.py#L17)):*
<details><summary><i>Previously, RSWilson1 (Robert Wilson) wrote…</i></summary><blockquote>

rescued always is associated with another filter i.e. LowSupport, Blacklist etc.. never on it's own so probably best to update these example scenarios which are just rescued.
</blockquote></details>

Updated


<!-- Sent from Reviewable.io -->

RSWilson1 reviewed (changes requested):
> @RSWilson1 reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/5)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @Arun-bioinformatics)

___
*[`DI-2086/correlation_plots_pdf_html.py` line 17 at r1](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/5#-Ob2EtQ5-P-tu0d5NP9S:-Ob2MZSX-YZOJqFLm82T:bwsk2y) ([raw file](https://github.com/eastgenomics/solid_cancer_requests/blob/90cf430665bade23458e04ce705efb4d7ed2183e/DI-2086/correlation_plots_pdf_html.py#L17)):*
<details><summary><i>Previously, Arun-bioinformatics (Arun Karnani Khemlani) wrote…</i></summary><blockquote>

Updated
</blockquote></details>

Could you update the rest?


<!-- Sent from Reviewable.io -->

Commit:
Removed all 'rescued' cases from the docstring in add_filter_column_change()

RSWilson1 reviewed (approved):
> @RSWilson1 reviewed 1 of 1 files at r3, all commit messages.
> @RSWilson1 dismissed @Arun-bioinformatics and @coderabbitai[bot] from a discussion.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/5)* status: all files reviewed, 1 unresolved discussion (waiting on @Arun-bioinformatics)



<!-- Sent from Reviewable.io -->

RSWilson1 reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/5)* status: all files reviewed, 2 unresolved discussions (waiting on @Arun-bioinformatics)



<!-- Sent from Reviewable.io -->

RSWilson1 reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/5)* status: all files reviewed, 2 unresolved discussions (waiting on @Arun-bioinformatics)



<!-- Sent from Reviewable.io -->

RSWilson1 reviewed (approved):
> @RSWilson1 dismissed @coderabbitai[bot] from 2 discussions.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/solid_cancer_requests/5)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @Arun-bioinformatics)



<!-- Sent from Reviewable.io -->
